boggle.js

```diff
@@ -6,26 +25,136 @@ function newBoggleBoard() {
     ['_', '_', '_', '_'],
   ];
 }
-
-/**
- * Prints out a boggle board.
- */
 function printBoggleBoard(boggleBoard) {
+  let isQ = false;
+  for (let i = 0; i < boggleBoard.length; i++) {
+    for (let j = 0; j < boggleBoard[i].length; j++) {
+      if (boggleBoard[i][j] === 'Q') {
+        boggleBoard[i][j] = 'Qu';
+        isQ = true;
+      }
+    }
+  }
+
+  if (isQ === true) {
+    for (let i = 0; i < boggleBoard.length; i++) {
+      for (let j = 0; j < boggleBoard[i].length; j++) {
+        boggleBoard[i][j] = boggleBoard[i][j].padEnd(2);
+      }
+    }
+  }
   for (let row of boggleBoard) {
     console.log(row.join(' '));
   }
 }
 
-/**
- * Shake a boggle board and fill it with letters.
- */
-function shake(boggleBoard) {
-  // This is your job. :)
+// function shake(boggleBoard) {
+// let abc = 'abcdefghijklmnopqrstuvwxyz';
+//   for(let row of boggleBoard){
+//     for(let i = 0; i < boggleBoard.length; i++){
+//       row[i] = abc[Math.floor(Math.random()*26)];
+//     }
+
+//   }
+//   return boggleBoard;
+
+// }
+function shake (boggleBoard) {
+  let dice = [
+    {
+      combo: 'AAEEGN',
+    },
+    {
+      combo: 'ELRTTY',
+    },
+    {
+      combo: 'AOOTTW',
+    },
+    {
+      combo: 'ABBJOO',
+    },
+    {
+      combo: 'EHRTVW',
+    },
+    {
+      combo: 'CIMOTU',
+    },
+    {
+      combo: 'DISTTY',
+    },
+    {
+      combo: 'EIOSST',
+    },
+    {
+      combo: 'DELRVY',
+    },
+    {
+      combo: 'ACHOPS',
+    },
+    {
+      combo: 'HIMNQU',
+    },
+    {
+      combo: 'EEINSU',
+    },
+    {
+      combo: 'EEGHNW',
+    },
+    {
+      combo: 'AFFKPS',
+    },
+    {
+      combo: 'HLNNRZ',
+    },
+    {
+      combo: 'DEILRX',
+    },
+
+  ];
+  for (let row of boggleBoard) {
+    for (let i = 0; i < boggleBoard.length; i++) {
+      let randomDice = Math.floor(Math.random() * 15);
```

Comment:
Don't hard-code `15`. What if we wanted to play super boggle, which is 6x6? You should be able to change `newBoggleBoard` to generate a different board and the rest of the code should work perfectly without modification.